pkg/provider/versions.go

```diff
@@ -0,0 +1,86 @@
+/*
```

Comment:
The new functions belong in the `pkg/cluster/cluster.go` and not here. Having said that, we need to do some refactoring and extract logic out of the cluster.go file.

Reply:
Thanks for calling this out.  I started using pkg/provider/ indiscriminately as a place to add new files without thinking about the meaning of "cluster" vs "provider"...
versions are not provider-specific, they are a quality of clusters, so make sense in cluster :+1:.

=> Moved to pkg/cluster/versions.go, PTAL

Reply:
in #181 I'll move some more stuff out of cluster.go: `getMachineTypes` and friends are going to new file pkg/provider/machine_types.go.